dts/bindings/clock/nordic,nrf-hfxo.yaml

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Nordic nRF high-frequency crystal oscillator
+
+compatible: "nordic,nrf-hfxo"
```

Comment:
@gmarull only issue with this is this is specific to that device, if this is ported to the nrf53 then this list of values is likely going to be wrong, how would that be handled?

Reply:
then we'll need different compatibles. let's figure that out once 53 is ported.